### PR TITLE
Added option to set custom SMB challenge in smbserver.py

### PR DESCRIPTION
**Pull request by byt3bl33d3r**
Hi!
As the title says, adds an option to set the SMB challenge in smbserver.py.

Big fan of this project btw! 
Un saluto!

### CODE REVIEW

**Comment by asolino** [comment]
tante grazie!

I added a few comments

File: impacket/smbserver.py [hunk 1]
```diff
@@ -4181,6 +4183,16 @@ def removeShare(self, shareName):
         self.__srvsServer.setServerConfig(self.__smbConfig)
         self.__server.processConfigFile()
         self.__srvsServer.processConfigFile()
+
+    def setSMBChallenge(self, challenge):
+        if challenge != '':
+            chal = ''
+            for i in range(0,len(challenge),2):
+                chal += challenge[i:i+2].decode("hex")
+
```
Comment (asolino) [inline impacket/smbserver.py]: Why not challenge.decode('hex') and that's it?

**Comment by byt3bl33d3r** [comment]
@asolino you are absolutly correct. have no idea what i was thinking lol

**Comment by asolino** [comment]
Thanks for the change mate!.. Just tested it and works great.. 
I just commented a small cosmetic change and clarification.

File: examples/smbserver.py [hunk 2]
```diff
@@ -54,6 +54,7 @@
 
     server.addShare(options.shareName.upper(), options.sharePath, comment)
     server.setSMB2Support(options.smb2support)
+    server.setSMBChallenge('') #here you can set a custom smb challenge, if empty defaults to 'A'*8 (remember: must be 8 bytes (16 chars) long)
```
Comment (asolino) [inline examples/smbserver.py]: I changed the comment a lil bit to make more clear. Let me know if you're ok with it and please change it (I also put it before the actual code):

    # Here you can set a custom SMB challenge in hex format
    # If empty defaults to '4141414141414141'
    # (remember: must be 16 hex bytes long)
    # e.g. server.setSMBChallenge('12345678abcdef00') 
    server.setSMBChallenge('') 

grazie!

Comment (byt3bl33d3r) [inline examples/smbserver.py]: looks good to me!

**Comment by asolino** [comment]
Just merged.. thanks a lot!

**Comment by byt3bl33d3r** [comment]
@asolino np! might see a couple of more coming your way over the next few weeks ;)

**Pull request merged**